### ISSUE TITLE
Use "current" in update URLs instead of CORE_COMPATIBILITY to retrieve module updates

### ISSUE THREAD

**Issue by weitzman** [drush-ops/drush#4310]
See https://github.com/drupal/drupal/commit/b126c02e2dc855bb7f57df70c36a662c156ceaa6

**Comment by greg-1-anderson** [comment]
Hm I guess we can use class_exists on `ModuleVersion` to use the new technique.

I would guess this is only for Drupal 9 -- i.e. we are not breaking `drush pm-update` for old versions of Drupal, I presume.

**Comment by greg-1-anderson** [comment]
I guess we could fork our own copy of `ModuleVersion` if this change is universal.

**Comment by greg-1-anderson** [comment]
Yeah, from [#3074993](https://www.drupal.org/node/3074993) it appears this is only for data returned from the `current` URL. I already partially fixed pm-* to use current, but it is not perfect. The techniques in the linked issue can improve the pm-* code for Drupal 9.
